### DISCUSSION TITLE
[SUPPORT]  Cookies not deleting

### DISCUSSION THREAD

**Discussion by Mr-Personality** [Cookie-AutoDelete/Cookie-AutoDelete#976] [Q&A / Support] [answered]
I've just done the following:

1. Installed Cookie AutoDelete for the first time
2. Visited my regular sites like email, and whitelisted them
3. Clicked the big yellow "Clean" button, expecting all non-whitelisted cookies to be deleted

But if I go to Options > Privacy & Security > Cookies and Site Data > Manage Data, there are many cookies still there that have not been whitelisted.

Am I doing something wrong?

**Comment by kennethtran93** [comment]
Moved to Discussions.

Are the entries remaining part of your list of opened tabs?  If not, what sites are remaining, and is it the cookie count that is not zero or that there may be some data remaining.  They do not necessarily correlate to one another.

**Reply by Mr-Personality** [reply]
> Are the entries remaining part of your list of opened tabs?

No. I do have quite a lot of opened tabs, but none of them match the cookies that are being retained. (Though tabs don't load until you click on them anyway, so none of them should be loading cookies.)

> If not, what sites are remaining

There are quite a few, including:
www.instagram.com
www.pinterest.co.uk
twitter.com
onedrive.live.com

> and is it the cookie count that is not zero or that there may be some data remaining. They do not necessarily correlate to one another.

I'm sorry, I don't understand this question. What is the 'cookie count' you speak of?

**Reply by kennethtran93** [reply]
> > Are the entries remaining part of your list of opened tabs?
> 
> No. I do have quite a lot of opened tabs, but none of them match the cookies that are being retained. (Though tabs don't load until you click on them anyway, so none of them should be loading cookies.)

Cookies are usually previously saved on browser/computer beforehand (i.e. when you visited them previously from that tab and most likely restarted the browser keeping those tabs intact/persist.)  Those still count as opened tabs.


> > If not, what sites are remaining
> 
> There are quite a few, including:
> [www.instagram.com](http://www.instagram.com)
> [www.pinterest.co.uk](http://www.pinterest.co.uk)
> twitter.com
> onedrive.live.com

Are these domains part of the tabs you have opened?  They don't have to be loaded to be part of opened tabs.


> > and is it the cookie count that is not zero or that there may be some data remaining. They do not necessarily correlate to one another.
> 
> I'm sorry, I don't understand this question. What is the 'cookie count' you speak of?

In Firefox, the popup of 'Manage Data' shows the following columns:
![image](https://user-images.githubusercontent.com/6724477/100534026-afa0d600-31bf-11eb-8477-73751b1260c0.png)

The cookie count is the 2nd column 'cookies', whereas the column after 'storage' is usually not cookie data but localstorage, indexedDB, and maybe sessionstorage.  Note that if the storage value is 64.0kb or 48.0kb, it may indicate an empty indexedDB.  That's usually fine, somebody else had brought it up and I believe I have a reference of it in either the FAQ or Documentation.

**Reply by Mr-Personality** [reply]
Kenneth, thank you for your help, and explanation.

The cookie count for some of them is not zero.

For example, www.instagram.com has 4 cookies. Yet I don't have an instagram account, and very rarely visit the site, and there are no open instagram tabs.

**Reply by kennethtran93** [reply]
Are you using any containers?

**Reply by Mr-Personality** [reply]
> Are you using any containers?

I use the add-on Facebook Container. I also use NoScript, Privacy Badger, and Ublock Origin. That's all.

**Reply by kennethtran93** [reply]
> > Are you using any containers?
> 
> I use the add-on Facebook Container. I also use NoScript, Privacy Badger, and Ublock Origin. That's all.

Do you have container cleanup enabled in CAD Settings? 

There may be a slight incompatibility when using Facebook Containers with CAD, according to various user reports.

**Reply by Mr-Personality** [reply]
"Enable Support for Firefox's Container Tabs" is enabled
"Enable Cleanup Log and Counter" is disabled

**Reply by kennethtran93** [reply]
I'll look into why Facebook Container is incompatible with CAD and what I can do about it, if any.  Until then, you'll either need to turn off Facebook Container or just keep those cookies there in the meantime.  From what other say in the issues area, the culprit was Facebook Container that prevented some data from being removed.

**Reply by kennethtran93** [reply]
A quick test on the developer edition shows nothing out of the ordinary - that is, based on cookie cleaning only.  Did not enable any of the other site data cleanup yet.  Checked `about:preferences#privacy` and `Manage Data...` just like you, confirmed that there was at least one cookie on 'www.instagram.com' (well at least a self-created cookie through that container), then performed the cleanup from the popup on a new tab.  Upon refresh of the preferences page (you need to do that to get updated figures), the cookie count column says zero.

If you're up for some debugging on your end, could you try checking the debug option, and follow the instructions to get the webextension console log panel showing, then clearing the log before reattempting the process for the cookie cleanup?  After a few seconds after the last event (or until at least a notification shows up about cleaning if enabled), could you copy-paste the log into another file and read over it to see what happened down the line and/or send it my way?  All cookie 'values' are sanitized.  The rest of the cookie details are left as is, as well as the tab info (which usually includes all opened domains).

I don't think the 'cleanup log and counter' plays any effect into this as those are usually post-cleanup actions.

As a side question, do you have firstpartyisolate enabled?

Also, were there any notification errors that appeared during cleanup?  Normally any errors raised would terminate the rest of the cleanup.

**Reply by Mr-Personality** [reply]
Thanks for the help Kenneth. I'll get back to you about this within the next couple of days.

**Reply by kennethtran93** [reply]
> 
> 
> Thanks for the help Kenneth. I'll get back to you about this within the next couple of days.

Just wanted to check in here - how is it looking on your end?

**Comment by kennethtran93** [comment, marked as answer]
Marking this as resolved due to the staleness of this discussion.  At this point I am presuming the issue was resolved.

**Comment by mnodx** [comment]
Hello Kenneth, I'm running into a similar issue.

I find that most if not all my cookies are persistent and it seems none ore being deleted on restart/tab close/unload as expected.

I have a simple Chrome installation running with no users signed in, and no other tools/extensions to manipulate cookies. Screenshot of my other active extensions below. 

I tried to troubleshoot by 'starting with a clean slate'.. I disable the extension 'EditThisCookie', I deleted all site data through Chrome, I removed & reinstalled the Cookie-AutoDelete extension and I set to 'Enable' all options under 'Automatic Cleaning Options'. I didn't set any white/grey lists. 

I then visited a few websites, 'ended chrome task' through Windows task manager and restarted my browser. Cookies still there. 

Having all the options under 'Automatic Cleaning Options' set to enable should delete all cookies after closing Chrome, correct? 

Are there any setting in Chrome that needs to be enabled/disabled for Cookie-AutoDelete to work properly?

I'd really love the features Cookie-AutoDelete is providing and I can't wait to have it working as expected. 

Thank you in advance!

---- 
- OS: x86-64 Windows (Please add OS version on paste)
- Browser Info: Chrome (Please add version number on paste)
- CookieAutoDelete Version: 3.8.2

- activeMode: true
- delayBeforeClean: 15
- discardedCleanup: true
- domainChangeCleanup: true
- enableGreyListCleanup: true
- cleanCookiesFromOpenTabsOnStartup: true
- cleanExpiredCookies: true
- siteDataEmptyOnEnable: true
- cacheCleanup: false
- indexedDBCleanup: false
- localStorageCleanup: false
- pluginDataCleanup: false
- serviceWorkersCleanup: false
- contextualIdentities: false
- contextualIdentitiesAutoRemove: true
- statLogging: true
- showNumOfCookiesInIcon: true
- keepDefaultIcon: false
- showNotificationAfterCleanup: true
- manualNotifications: true
- notificationOnScreen: 3
- enableNewVersionPopup: false
- sizePopup: 16
- sizeSetting: 16
- contextMenus: true
- debugMode: false
![image](https://github.com/Cookie-AutoDelete/Cookie-AutoDelete/assets/102597707/e86a61be-3081-4789-83c4-5844726c1851)

**Reply by mnodx** [reply]
*Chrome  121.0.6167.161 (Official Build) (64-bit)
*OS version Windows 11 home 22631.3007
